### PR TITLE
Add docs for configuring SSO via Azure AD

### PR DESCRIPTION
**Pull request by yob**
Azure AD is a reasonably common SSO option used by our customers. It uses SAML and folks have worked out how to get it going via the "Custom SAML provider" option, so this is just documenting a thing that already works.

Azure AD is a cloud hosted identity service from Microsoft. Many organizations who use a windows stack are starting to migrate from a traditional on-prem Active Directory to Azure AD.

I put these docs together by logging into our test Azure account and clicking around until I had a working setup.

### CODE REVIEW

File: pages/integrations/sso/azure_ad.md.erb [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    * Delete all remaining mappings except for the 4 listed in the screenshot below:
+    * Click 'Save'
+    * Return to provisioning setttings
+1. Toggle 'Provisioning Status' to 'On'
+1. Click 'Save'
+1. Return to the `Provisioning` menu of the enterprise app and view the 'Current cycle status' section.
+    * If provisioning is working, this should say 'Initial cycle completed'
+    * If errors are displayed, click 'View provisioning logs' and for further details on what went wrong
+
+<%= image "azuread-user-mappings.png", width: 1440/2, height: 548/2, alt: "Screenshot of the required user mappings on Azure" %>
```
Comment (yob) [inline pages/integrations/sso/azure_ad.md.erb:110]: The positioning of this image is a problem. It relates to a step in the middle of the above list ("Customise the User mappings").

I'm not sure we can convinced markdown to let us insert it in the middle of a list though?

Comment (plaindocs) [reply pages/integrations/sso/azure_ad.md.erb:110]: This should work, but has some super odd spacing 

```
diff --git a/pages/integrations/sso/azure_ad.md.erb b/pages/integrations/sso/azure_ad.md.erb
index a89f04a5..9cdad4d8 100644
--- a/pages/integrations/sso/azure_ad.md.erb
+++ b/pages/integrations/sso/azure_ad.md.erb
@@ -95,6 +95,7 @@ Go to your Buildkite application in AzureAD to set up provisioning:
     * Toggle 'Enabled' to 'No' and click 'Save'
     * Return to provisioning setttings
 1. Customise the User mappings
+    <%= image "azuread-user-mappings.png", width: 1440/2, height: 548/2, alt: "Screenshot of the required user mappings on Azure" %>
     * Expand 'Mappings'
     * Click 'Provision Azure Active Directory Users'
     * Click the first row of the 'Attribute Mappings' table ('userPrincipalName'), change the 'Source attribute' to 'mail' and click 'OK'
@@ -107,5 +108,4 @@ Go to your Buildkite application in AzureAD to set up provisioning:
     * If provisioning is working, this should say 'Initial cycle completed'
     * If errors are displayed, click 'View provisioning logs' and for further details on what went wrong
 
-<%= image "azuread-user-mappings.png", width: 1440/2, height: 548/2, alt: "Screenshot of the required user mappings on Azure" %>
```

File: pages/integrations/sso/azure_ad.md.erb [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    * Expand 'Mappings'
+    * Click 'Provision Azure Active Directory Groups'
+    * Toggle 'Enabled' to 'No' and click 'Save'
+    * Return to provisioning setttings
+1. Customise the User mappings
+    * Expand 'Mappings'
+    * Click 'Provision Azure Active Directory Users'
+    * Click the first row of the 'Attribute Mappings' table ('userPrincipalName'), change the 'Source attribute' to 'mail' and click 'OK'
+    * Delete all remaining mappings except for the four listed in the following screenshot:
+    <%= image "azuread-user-mappings.png", width: 1440/2, height: 548/2, alt: "Screenshot of the required user mappings on Azure" %>
```
Comment (plaindocs) [inline pages/integrations/sso/azure_ad.md.erb:102]: What I don't know is where this is coming from, which looks like the odd spacing?

```
<div style="max-width: 720px"><div class="responsive-image-container" style="padding-bottom: 38.05555555555556%">
```

**Comment by plaindocs** [comment]
- [ ] There are a couple of TODO urls
- [ ] The list + image spacing could be improved

**Comment by yob** [comment]
> Can you configure SCIM with GraphQL? It's mostly on the Azure side, so probably not? Should we mention it over on the GraphQL side? Probably

There's no setup to do on our side for SCIM, so no. It's all on the Azure side.

**Comment by yob** [comment]
I've cherry-picked @lachlanhardy's fix for the image spacing from #1154.

Thanks so much for your help @plaindocs and @lachlanhardy! This is technically accurate and good to go from our perspective :+1:

**Review by plaindocs — APPROVED** [review]

**Pull request merged**